crates/rome_js_analyze/src/react.rs

```
@@ -40,40 +112,148 @@ impl ReactCreateElementCall {
+    fn find_prop_by_name(&self, prop_name: &str) -> Option<JsPropertyObjectMember> {
+        self.new_props.as_ref().and_then(|props| {
+            let members = props.members();
+            members.iter().find_map(|member| {
+                let member = member.ok()?;
+                let property = member.as_js_property_object_member()?;
+                let property_name = property.name().ok()?;
+
+                let property_name = property_name.as_js_literal_member_name()?;
+                if property_name.name().ok()? == prop_name {
+                    Some(property.clone())
+                } else {
+                    None
+                }
+            })
+        })
+    }
+}
+
+/// List of valid [`React` API]
 ///
-/// ```js
-/// createElement()
-/// ```
+/// [`React` API]: https://reactjs.org/docs/react-api.html
+const VALID_REACT_API: [&str; 14] = [
+    "Component",
+    "PureComponent",
+    "memo",
+    "createElement",
+    "cloneElement",
+    "createFactory",
+    "isValidElement",
+    "Fragment",
+    "createRef",
+    "forwardRef",
+    "lazy",
+    "Suspense",
+    "startTransition",
+    "Children",
+];
+
+/// Checks if the current [JsCallExpression] is a potential [`React` API].
+/// The function has accepts a `api_name` to check against
 ///
-/// The logic of this second case is very similar to the previous one, simply the node that we have
-/// to inspect is different.
-pub(crate) fn is_react_create_element(
-    call_expression: &JsCallExpression,
+/// [`React` API]: https://reactjs.org/docs/react-api.html
+pub(crate) fn is_react_call_api(
+    expression: &JsAnyExpression,
     model: &SemanticModel,
-) -> Option<ReactCreateElementCall> {
-    let callee = call_expression.callee().ok()?;
-    let is_react_create_element = match callee {
+    api_name: &str,
+) -> Option<bool> {
+    // we bail straight away if the API doesn't exists in React
+    if !VALID_REACT_API.contains(&api_name) {
```

This could be a `debug_assert!`, this way the check isn't performed on release builds, and panics straight away on debug builds if the provided name isn't valid instead of silently failing